distribution/openrct2.d.ts

```diff
@@ -4456,4 +4577,81 @@ declare global {
         readonly parents: number[];
         readonly children: number[];
     }
+
+    interface ObjectManager {
+        /**
+         * Gets all the objects that are installed and can be loaded into the park.
+         */
+        readonly installedObjects: InstalledObject[];
+
+        /**
+         * Gets the installed object with the given identifier, or null
+         * if the object was not found.
+         * @param identifier The object identifier.
+         */
+        getInstalledObject(identifier: string): InstalledObject | null;
+
+        /**
+         * Attempt to load the object into the current park at the given index for the object type.
+         * If an object already exists at the given index, that object will be unloaded and this object
+         * will replace it, providing the object type is the same.
+         * @param identifier The object identifier.
+         * @param index The index to load the object to. If not provided, an empty slot will be used.
+         * @returns The index of the loaded object.
+         */
+        load(identifier: string, index?: number): LoadedObject | null;
+
+        /**
+         * Attempt to load the given objects into the current park, given they are not already loaded.
+         */
+        load(identifiers: string[]): (LoadedObject | null)[];
+
+        /**
+         * Unloads the object, if loaded.
+         * @param identifier The object identifier to unload.
+         */
+        unload(identifier: string): void;
+
+        /**
+         * Unloads the specified objects, if loaded.
+         * @param identifiers The object identifiers to unload.
+         */
+        unload(identifiers: string[]): void;
```

Comment:
What will happen if the object is in use in the park?

Reply:
It copes most of the time. If it doesn't, it is a bug in the OpenRCT2 code. I tried unloading a merry go round, and all that happened was the vehicle disappears - but the ride remains with the floor.

Reply:
So it is preferable that it is possible to unload objects that are still in use in the park?

Reply:
Well, it is already possible to change rides and tile elements to point to a completely empty object slot. I didn't intend for the plugin API to have constraints like this as plugins may exist that are there to hack or unhack parks. That is why I also allow loading a different object in place of an existing one, it allows you to swap a footpath with a different one for example.